### PR TITLE
fix(compiler): handle null window.location.origin

### PR DESCRIPTION
**Pull request by brunor97**
when using a data URI or file URI (#2582), or when using iframe's `srcdoc`, the value for window.location.origin can be `null`. In these particular cases, passing `null` to `new URL(url, null)` will throw the following error: Failed to construct 'URL': Invalid base URL

### CODE REVIEW

**Comment by 00salmon** [comment]
+1, currently blocked by this when using components in an iframe

**Comment by brunor97** [comment]
> +1, currently blocked by this when using components in an iframe

As a hotfix, I'm using patch-package to apply this fix on my project's side, while we wait for this PR to be merged and published 

Not a long term solution but hopefully this can unblock you 😃

**Comment by 00salmon** [comment]
> > +1, currently blocked by this when using components in an iframe
> 
> As a hotfix, I'm using patch-package to apply this fix on my project's side, while we wait for this PR to be merged and published
> 
> Not a long term solution but hopefully this can unblock you 😃

that's more convenient than editing the file in `dist` directly before each publish, thank you!

**Comment by brunor97** [comment]
> > > +1, currently blocked by this when using components in an iframe
> > 
> > 
> > As a hotfix, I'm using patch-package to apply this fix on my project's side, while we wait for this PR to be merged and published
> > Not a long term solution but hopefully this can unblock you 😃
> 
> that's more convenient than editing the file in `dist` directly before each publish, thank you!

No problem! Glad I could be of help 😄

**Comment by Sanderand** [comment]
+1

**Comment by loganvolkers** [comment]
Bump @rwaskiewicz

If this needs some changes to get it over the line, please let me know. We have dozens of stencil projects that all need this fix manually applied so they are compatible with sandboxed iframes and `srcdoc` iframes

**Comment by loganvolkers** [comment]
e.g. I'm happy to add some test cases to this PR if that will help it get across the line

**Comment by jan-paulus** [comment]
Any update on this?

**Comment by Sanderand** [comment]
Any update on this?

**Comment by leonheess** [comment]
@rwaskiewicz, please merge these changes or outline any requirements that need to be fulfilled

**Comment by jan-paulus** [comment]
@rwaskiewicz Could you take a look?

We've used `patch-package` to workaround this issue for more than a year now.

File: src/compiler/output-targets/dist-lazy/generate-system.ts
Comment (rwaskiewicz) [inline src/compiler/output-targets/dist-lazy/generate-system.ts:100]:
```suggestion
      var url = new URL('${corePath}', new URL(resourcesUrl, window.location.origin !== 'null' ? window.location.origin : undefined));
```

**Review by rwaskiewicz — COMMENTED** [review]
LGTM - I have one stylistic nitpick that I think can be applied in GH

**Review by rwaskiewicz — APPROVED** [review]
I'm going to be merging this as an admin - the failing CI step is a known issue when running against forks.  The step in question is our tech debt burndown chart, which I'm able to see the output of (no new unused functions or strict null checks violations are added in this PR)

**Pull request merged**
